src/lsp/main.ts

```diff
@@ -788,17 +788,18 @@ export async function getClojuredocs(
 }
 
 // TODO: This feels a bit brute, what are other ways to wait for the client to initialize?
-function getClient(timeout: number): Promise<LanguageClient> | undefined {
+function getClient(timeout: number): Promise<LanguageClient> {
     const start = Date.now();
     return new Promise(waitForClientStarted);
 
     function waitForClientStarted(resolve, reject) {
-        if (getStateValue(LSP_CLIENT_KEY)) {
-            resolve(getStateValue(LSP_CLIENT_KEY));
-        } else if (getStateValue(LSP_CLIENT_KEY_ERROR)) {
-            reject(
-                new Error('clojure-lsp: ' + getStateValue(LSP_CLIENT_KEY_ERROR))
-            );
+        const client = getStateValue(LSP_CLIENT_KEY);
+        const error = getStateValue(LSP_CLIENT_KEY_ERROR);
+
+        if (client) {
+            resolve(client);
+        } else if (error) {
+            reject(new Error('clojure-lsp: ' + error));
```

Comment:
this section is just a refactoring to prevent multiple fetches and not repeat ourselves as much. plus it makes the condition branching here a lot clearer